Sources/Transaction/Memos/SenderMemo.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  Copyright (c) 2020-2021 MobileCoin. All rights reserved.
+//
+
+import Foundation
+
+
+struct SenderMemo {
+    let memoData: Data64
+    let addressHash: AddressHash
+}
+
+struct RecoverableSenderMemo {
+    let memoData: Data64
+    let addressHash: AddressHash
+    let accountKey: AccountKey
```

Comment:
If TxOuts and memos are going to be cached, then this account key will too. We need to make sure AccountKeys won't be stored insecurely.